File: frontend/src/domain/entities/CV.ts
export interface Experience {
  period: string;
  role: string;
  company: string;
  companyUrl?: string;
  location?: string;
  workMode?: string;
  description: string;
  background?: string;
}

export interface ContactInfo {
  email: string;
  phone: string;
  location: string;
  willingnessToTravel: string;
}

export interface SkillSubCategory {
  name: string;
  items: string[];
}

export interface SkillCategory {
  name: string;
  subCategories: SkillSubCategory[];
}

export interface Education {
  degree: string;
  institution: string;
  notes: string;
}

export interface Certification {
  category: string;
  title: string;
  issuer: string;
}

export interface CV {
  name: string;
  lastName: string;
  title: string;
  summary: string;
  contactInfo?: ContactInfo | null;
  experiences: Experience[];
  skillCategories: SkillCategory[];
  education: Education[];
  certifications: Certification[];
}

File: backend/src/Domain/Entities/CV.cs
namespace Backend.Domain.Entities;

public sealed class Experience
{
    public required string Period { get; init; }
    public required string Role { get; init; }
    public required string Company { get; init; }
    public string CompanyUrl { get; init; } = string.Empty;
    public string Location { get; init; } = string.Empty;
    public string WorkMode { get; init; } = string.Empty;
    public required string Description { get; init; }
    public string Background { get; init; } = string.Empty;
}

public sealed class CV
{
    public required string Name { get; init; }
    public required string LastName { get; init; }
    public required string Title { get; init; }
    public required string Summary { get; init; }
    public ContactInfo? ContactInfo { get; init; }
    public required IReadOnlyList<Experience> Experiences { get; init; }
    public required IReadOnlyList<SkillCategory> SkillCategories { get; init; }
    public required IReadOnlyList<Education> Education { get; init; }
    public required IReadOnlyList<Certification> Certifications { get; init; }
}
